manager-queue.js

```diff
@@ -9,17 +9,24 @@ class QueueManager {
         const { pullRequestNumber } = item;
 
         return redis.sismember(`${prefix}-index:${owner}:${repo}`, { pullRequestNumber })
-            .then(isPresent => redis.sadd(`${prefix}-index:${owner}:${repo}`, { pullRequestNumber })
-                .then(() => isPresent ? '' : redis.push(`${prefix}:${queue}`, item)))
-            .then(() => {
-                postal.publish({
-                    channel: 'queue',
-                    topic: 'item.add',
-                    data: {
-                        queue,
-                        item,
-                    },
-                });
+            .then(isPresent => {
+                if (!isPresent) {
+                    redis.sadd(`${prefix}-index:${owner}:${repo}`, { pullRequestNumber })
+                        .then(() => redis.push(`${prefix}:${queue}`, item))
+                        .then(queueLength => {
+                            postal.publish({
+                                channel: 'queue',
+                                topic: 'item.add',
+                                data: {
+                                    queue,
+                                    item,
+                                    index: queueLength - 1,
+                                },
+                            });
+                        });
+                } else {
+                    throw new Error('Item already present in the queue');
```

Comment:
I prefer this to fail silently (log if you want to) to avoid service disruption

Reply:
I'll go with the log then, thanks for pointing that out.